hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -618,11 +621,14 @@ public List<String> getMetadataPartitionsInflight() {
     );
   }
 
-  public List<String> getMetadataPartitions() {
-    return StringUtils.split(
-        getStringOrDefault(TABLE_METADATA_PARTITIONS, StringUtils.EMPTY_STRING),
-        CONFIG_VALUES_DELIMITER
-    );
+  public Set<String> getMetadataPartitions() {
+    if (metadataPartition == null) {
+      metadataPartition = new HashSet<>(
+              StringUtils.split(getStringOrDefault(TABLE_METADATA_PARTITIONS, StringUtils.EMPTY_STRING),
+                      CONFIG_VALUES_DELIMITER)
+      );
```

Comment:
This is not thread-safe.  Could the method just return a new HashSet instance each time?